### PR TITLE
Doc with matrix of implemented scenarios per repo

### PR DESCRIPTION
**Pull request by aabmass**
This PR includes a Go script to generate the markdown and the generated markdown. It queries GCP APIs and parses statuses from build logs of the most recent successful build of each trigger.

Automation can be made to re-build the `matrix.md` file periodically.

### CODE REVIEW

File: cmd/testmatrix/main.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+		panic(err)
+	}
+
+	var wg sync.WaitGroup
+	results := make([]*result, len(listTriggersRes.Triggers))
+	for i, trigger := range listTriggersRes.Triggers {
+		wg.Add(1)
+		go handleTrigger(
+			ctx,
+			&wg,
```
Comment (tbarker25) [inline cmd/testmatrix/main.go:123]: I don't think we should be passing in the wait group as an argument. It's less error-prone to keep synchronization as the caller's responsibility. I can already see a deadlock here - handleTrigger has an early return that doesn't call wg.Done().

```go
var wg sync.WaitGroup
for ... {
  wg.Add(1)
  go func() {
    handleTrigger(...)
    wg.Done()
  }
}
wg.Wait()
```

File: cmd/testmatrix/main.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+	}
+
+	// fetch the latest successful build
+	listRes, err := cloudbuildService.Projects.Builds.List(projectId).
+		Context(ctx).
+		Filter(fmt.Sprintf(`trigger_id="%v" AND status="SUCCESS"`, trigger.Id)).
+		PageSize(1).
+		Do()
+	if err != nil {
+		panic(err)
```
Comment (tbarker25) [inline cmd/testmatrix/main.go:201]: Consider returning an error and using https://pkg.go.dev/golang.org/x/sync/errgroup in the caller.

File: cmd/testmatrix/main.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+	}
+}
+
+// handleTrigger populates the resPtr with the results of the latest build of
+// the given trigger.
+func handleTrigger(
+	ctx context.Context,
+	wg *sync.WaitGroup,
```
Comment (tbarker25) [inline cmd/testmatrix/main.go:178]: See above: it's less error prone to remove the wg argument and have the caller handle synchronization.

File: cmd/testmatrix/main.go [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+	projectId string,
+	resPtr **result,
```
Comment (tbarker25) [inline cmd/testmatrix/main.go:180]: Can we return result instead?

File: cmd/testmatrix/main.go [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+	trigger *cloudbuild.BuildTrigger,
+	cloudbuildService *cloudbuild.Service,
+	storageClient *storage.Client,
+) {
+	if !triggerNameRe.MatchString(trigger.Name) {
+		return
```
Comment (tbarker25) [inline cmd/testmatrix/main.go:186]: This will deadlock because we aren't calling wg.Done()

File: cmd/testmatrix/main.go [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+	arg.MustParse(&args)
+
+	ctx := context.Background()
+	cloudbuildService, err := cloudbuild.NewService(ctx)
+	if err != nil {
+		panic(err)
+	}
+	storageClient, err := storage.NewClient(ctx)
+	if err != nil {
+		log.Fatal(err)
```
Comment (tbarker25) [inline cmd/testmatrix/main.go:106]: Is there a reason we aren't consistently calling either panic() or log.Fatal()?

**Review by tbarker25 — CHANGES_REQUESTED** [review]

File: cmd/testmatrix/main.go (hunk 1 above)
Comment (aabmass) [reply cmd/testmatrix/main.go:123]: Shamelessly copied it from here https://gobyexample.com/waitgroups. I agree this way makes more sense, thanks

File: cmd/testmatrix/main.go (hunk 2 above)
Comment (aabmass) [reply cmd/testmatrix/main.go]: Looks much better, thanks

File: cmd/testmatrix/main.go (hunk 6 above)
Comment (aabmass) [reply cmd/testmatrix/main.go]: Switched to just use panic in main so defers will run if they are added

**Review by tbarker25 — APPROVED** [review]

**Pull request merged**